content/client-lib-development-guide/features.textile

```diff
@@ -870,6 +871,7 @@ h4. Message
 ** @(TM2g)@ @name@ string
 ** @(TM2d)@ @data@ string, buffer or JSON-encodable object or array
 ** @(TM2e)@ @encoding@ string
+** @(TM2i)@ @extras@ JSON-like object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-like objects or JSON-like arrays. For context, the @extras@ field is planned for us with push notifications as a payload in the @push@ key, and later to allow arbitrary headers in the @header@ key that may be used for routing, filtering or to expose to consumers when retrieving messages from queues
```

Comment:
> . So it's maybe cleaner to keep things meaningful to us in a field (extras) and things meaningful to the clients in another (data).

I agree with that, but I am not sure that really affects the spec, it only applies to user documentation surely?